src/coreclr/nativeaot/System.Private.CoreLib/src/System/RuntimeExceptionHelpers.cs

```diff
@@ -109,7 +107,7 @@ public class RuntimeExceptionHelpers
                         return new AmbiguousImplementationException();
 
                     default:
-                        FailFast("The runtime requires an exception for a case that this class library does not understand.");
+                        FailFastInternal("The runtime requires an exception for a case that this class library does not understand.");
```

Comment:
How would you describe the distinction between internal FailFast and regular FailFast? It seems arbitrary. We use regular FailFast throughout the runtime and libraries. 

For example, failfasts from thread pool are regular fails fasts - why they are not internal fail fasts? https://github.com/dotnet/runtime/blob/e80ef8638b6d3f42add1bb046e032b2eff417473/src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.GateThread.cs#L248 .

Maybe we do not need the internal failfast.

If we want to have a dedicated code for access violations in managed code because they are special, we should name it accordingly.

Reply:
I'm going to back out the FailFastInternal change.